pkg/webhooks/machine_webhook.go

```diff
@@ -1426,3 +1478,184 @@ func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProv
 func isDeleting(obj metav1.Object) bool {
 	return obj.GetDeletionTimestamp() != nil
 }
+
+func defaultPowerVS(m *machinev1beta1.Machine, config *admissionConfig) (bool, []string, utilerrors.Aggregate) {
+	klog.V(3).Infof("Defaulting PowerVS providerSpec")
+
+	var errs []error
+	var warnings []string
+	providerSpec := new(machinev1.PowerVSMachineProviderConfig)
+	if err := unmarshalInto(m, providerSpec); err != nil {
+		errs = append(errs, err)
+		return false, warnings, utilerrors.NewAggregate(errs)
+	}
+	if providerSpec.UserDataSecret == nil {
+		providerSpec.UserDataSecret = &machinev1.PowerVSSecretReference{Name: defaultUserDataSecret}
+	}
+	if providerSpec.CredentialsSecret == nil {
+		providerSpec.CredentialsSecret = &machinev1.PowerVSSecretReference{Name: defaultPowerVSCredentialsSecret}
+	}
+	if providerSpec.SystemType == "" {
+		providerSpec.SystemType = defaultPowerVSSysType
+	}
+	if providerSpec.ProcessorType == "" {
+		providerSpec.ProcessorType = defaultPowerVSProcType
+	}
+	// Doubt: should we make Processors as pointer for easy validation
+	if providerSpec.Processors.IntVal == 0 || providerSpec.Processors.StrVal == "" {
+		providerSpec.Processors = intstr.IntOrString{Type: intstr.String, StrVal: "0.5"}
+	}
+	// Doubt: should we make MemoryGiB as pointer for easy validation
```

Comment:
We don't allow a zero value so we can assume that if it's set to zero, we are going to default it, I think this is fine as is